exercises/triangle/spec/triangle_spec.cr

```diff
@@ -2,59 +2,79 @@ require "spec"
 require "../src/*"
 
 describe "Triangle" do
-  it "true if all sides are equal" do
+  it "all sides are equal" do
     Triangle.new([2, 2, 2]).equilateral?.should eq(true)
   end
 
-  pending "false if any side is unequal" do
+  pending "any side is unequal" do
     Triangle.new([2, 3, 2]).equilateral?.should eq(false)
   end
 
-  pending "false if no sides are equal" do
+  pending "no sides are equal" do
     Triangle.new([5, 4, 6]).equilateral?.should eq(false)
   end
 
-  pending "All zero sides are illegal, so the triangle is not equilateral" do
+  pending "all zero sides is not a triangle" do
     Triangle.new([0, 0, 0]).equilateral?.should eq(false)
   end
 
-  pending "true if last two sides are equal" do
+  pending "sides may be floats" do
+    Triangle.new([0.5, 0.5, 0.5]).equilateral?.should eq(true)
+  end
+
+  pending "last two sides are equal" do
     Triangle.new([3, 4, 4]).isosceles?.should eq(true)
   end
 
-  pending "true if first two sides are equal" do
+  pending "first two sides are equal" do
     Triangle.new([4, 4, 3]).isosceles?.should eq(true)
   end
 
-  pending "true if first and last sides are equal" do
+  pending "first and last sides are equal" do
     Triangle.new([4, 3, 4]).isosceles?.should eq(true)
   end
 
   pending "equilateral triangles are also isosceles" do
     Triangle.new([4, 4, 4]).isosceles?.should eq(true)
   end
 
-  pending "false if no sides are equal" do
+  pending "no sides are equal" do
     Triangle.new([2, 3, 4]).isosceles?.should eq(false)
   end
 
-  pending "Sides that violate triangle inequality are not isosceles, even if two are equal" do
+  pending "first triangle inequality violation" do
     Triangle.new([1, 1, 3]).isosceles?.should eq(false)
   end
 
-  pending "true if no sides are equal" do
+  pending "second triangle inequality violation" do
+    Triangle.new([1, 3, 1]).isosceles?.should eq(false)
+  end
+
+  pending "third triangle inequality violation" do
+    Triangle.new([3, 1, 1]).isosceles?.should eq(false)
+  end
+
+  pending "sides may be floats" do
+    Triangle.new([0.5, 0.4, 0.5]).isosceles?.should eq(true)
+  end
+
+  pending "no sides are equal" do
     Triangle.new([5, 4, 6]).scalene?.should eq(true)
   end
 
-  pending "false if all sides are equal" do
+  pending "all sides are equal" do
     Triangle.new([4, 4, 4]).scalene?.should eq(false)
   end
 
-  pending "false if two sides are equal" do
+  pending "two sides are equal" do
     Triangle.new([4, 4, 3]).scalene?.should eq(false)
   end
 
-  pending "Sides that violate triangle inequality are not scalene, even if they are all different" do
+  pending "may not violate triangle inequality" do
     Triangle.new([7, 3, 2]).scalene?.should eq(false)
   end
+
+  pending "sides may be floats" do
```

Comment:
As discussed in the individual commit messages, the test group descriptions such https://github.com/exercism/problem-specifications/blob/master/exercises/triangle/canonical-data.json#L152 are not represented in this test file.

Therefore, there are three cases all named "sides may be floats"

However, the test descriptions have always been missing this context, so fixing this is not within the scope of this PR (and is outside of my interests). For example, exercise `forth` also is affected.